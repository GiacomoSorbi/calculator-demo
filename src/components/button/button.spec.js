import * as React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Button from './'

describe('Button should', () => {
  it('exist', () => {
    expect(Button).toBeDefined()
  })

  it('have a consistent structure', () => {
    const wrapper = mount(<Button onClick={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('trigger the callback function everytime the button is triggered', () => {
    const callback = jest.fn()
    const wrapper = mount(<Button onClick={callback} />)
    const button = wrapper.find('button')
    button.simulate('click')
    expect(callback).toHaveBeenCalledTimes(1)
    button.simulate('click')
    button.simulate('click')
    expect(callback).toHaveBeenCalledTimes(3)
  })
})
