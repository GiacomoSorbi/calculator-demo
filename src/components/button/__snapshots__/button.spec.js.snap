// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should have a consistent structure 1`] = `
<Button
  onClick={[MockFunction]}
>
  <button
    className="calculator-btn"
    onClick={[MockFunction]}
  />
</Button>
`;
